PubNub/Core/PubNub+ChannelGroup.m

```diff
@@ -104,31 +109,42 @@ -(void)fetchChannelsForChannelGroupWithRequest:(PNChannelGroupFetchRequest *)use
     PNGroupChannelsAuditCompletionBlock block = [handlerBlock copy];
     PNParsedRequestCompletionBlock handler; 
 
-    if (userRequest.operation == PNChannelGroupsOperation) {
-        PNLogAPICall(self.logger, @"<PubNub::API> Request channels for '%@' channel group.", userRequest.channelGroup);
-    } else PNLogAPICall(self.logger, @"<PubNub::API> Request channel groups list.");
-
     PNWeakify(self);
     handler = ^(PNTransportRequest *request, id<PNTransportResponse> response, __unused NSURL *location,
                 PNOperationDataParseResult<PNChannelGroupChannelsResult *, PNErrorStatus *> *result) {
         PNStrongify(self);
 
-        if (result.status.isError) {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-            result.status.retryBlock = ^{
-                [self fetchChannelsForChannelGroupWithRequest:userRequest completion:block];
-            };
-#pragma clang diagnostic pop
+        if (!result.status.isError) {
+            [self.logger debugWithLocation:@"PubNub" andMessageFactory:^PNLogEntry * {
+                NSUInteger count = (result.result.data.groups ?: result.result.data.channels).count;
+                NSString *message;
+                if (userRequest.operation == PNChannelGroupsOperation)
+                    message = PNStringFormat(@"List all channel groups success. Received %@ groups.", @(count));
+                else message = PNStringFormat(@"List channel group channels success. Received %@ channels.", @(count));
+                
+                return [PNStringLogEntry entryWithMessage:message];
+            }];
         }
 
         [self callBlock:block status:NO withResult:result.result andStatus:result.status];
     };
+                                        
+    [self.logger debugWithLocation:@"PubNub" andMessageFactory:^PNLogEntry * {
+        if (userRequest.operation == PNChannelGroupsOperation)
+            return [PNStringLogEntry entryWithMessage:@"List all channel groups."];
+        return [PNDictionaryLogEntry entryWithMessage:[userRequest dictionaryRepresentation]
+                                              details:@"List channel group channels with parameters:"];
+    }];
 
     [self performRequest:userRequest withParser:responseParser completion:handler];
 }
 
 - (void)channelsForGroup:(NSString *)group withCompletion:(PNGroupChannelsAuditCompletionBlock)block {
+    [self.logger warnWithLocation:@"PubNub" andMessageFactory:^PNLogEntry * {
```

Comment:
Should we log deprecated method calls? The caller will already receive a warning message in the IDE (Xcode)

Reply:
I was struggling with this decision. It is possible to silence any kind of IDE warnings in settings or during `pod install` in `Podfile`. If it is excessive, I'll remove it.

Reply:
I’ll leave it up to you. Personally, I’d skip these logs, but feel free to keep them if you think they’re useful